src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/ComponentVersionTelemetryInitializer.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace Microsoft.ApplicationInsights.AspNet.TelemetryInitializers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Channel;
+    using DataContracts;
+    using Microsoft.AspNet.Http;
+    using Microsoft.Extensions.Configuration;
+
+    /// <summary>
+    /// A telemetry initializer that populates telemetry.Context.Component.Version to the value read from project.json
+    /// </summary>
+    public class ComponentVersionTelemetryInitializer : TelemetryInitializerBase
+    {
+        private const string versionConfigurationOption = "dependencies:Microsoft.ApplicationInsights.AspNet";
+
+        public ComponentVersionTelemetryInitializer(IHttpContextAccessor httpContextAccessor):base(httpContextAccessor)
```

Comment:
I have made the change, but before creating a new Pull request. I wanted to ask if the approach was correct. 

I added this to the AddApplicationInsightsTelemetry method:

```
var projectJsonConfiguration = new ConfigurationBuilder().AddJsonFile("project.json").Build();
services.AddInstance<IConfiguration>(projectJsonConfiguration);
```

And Injecting the IConfiguration dependency on the constructor for the created class. 

It does not conflict with the TelemetryConfiguration but I don't feel it is the right place to have that service declared.